### DISCUSSION TITLE
Started...

### DISCUSSION THREAD

**Discussion by KnifMelti** [Romanitho/Winget-AutoUpdate#256] [Show and tell]
![image](https://user-images.githubusercontent.com/102996177/209719979-12da00db-3845-405a-b2d1-c441c45d1b7f.png)
![image](https://user-images.githubusercontent.com/102996177/209720594-fd68cfbd-e8f4-4671-8254-f45a6a47ac4e.png)

**Comment by KnifMelti** [comment]
#257

**Reply by KnifMelti** [reply]
Final:
![image](https://user-images.githubusercontent.com/102996177/212422844-9366c6aa-ee7a-490f-97f0-bffb5ab146ab.png)

**Reply by Romanitho** [reply]
haha fantastic ! Goof job ! 👏

**Reply by KnifMelti** [reply]
For **Intune** (maybe): **Group Policy Analytics**/[Create a Settings Catalog policy using your imported GPOs in Microsoft Intune (public preview)](https://learn.microsoft.com/en-us/mem/intune/configuration/group-policy-analytics-migrate)

**Reply by Romanitho** [reply]
even this: https://learn.microsoft.com/en-us/mem/intune/configuration/administrative-templates-import-custom

**Reply by Weatherlights** [reply]
I have a small idea here since I built a somewhat similar solution based on this project (Have not recognized that there was KnifMelti was already building a management overlay). But instead of providing a White-/Blacklist as an URL I added a control so you can enter the list within the policy itself. This way you avoid hosting the list somewhere outside of your management solution.

Also another idea I implemented (regarding the use within Intune) was making a copy of the logfile under %WINDOWS%\temp%COMPUTERNAME%-Winget-Autoconfiguration.log. This way you can gather the log files using the built in Intune Gather Diagnostics feature (so it is available for support purpose).

**Reply by KnifMelti** [reply]
I'm a dinosaur ;) (@Romanitho) so the _Intune Gather Diagnostics feature_ is nothing I'm familiar with...
...regarding "../added a control so you can enter the list within the policy itself/.."
I absolutely see the use of it as a fourth choice (URL/UNC/Local/GPO (REG)) and your fork [WinGet-AutoUpdate-Configurator](https://github.com/Weatherlights/Winget-AutoUpdate-Intune#winget-autoupdate-configurator) looks really nice (could have been implemented in this repo.. ..ah, I see: **WinGet-Autoupdate-Configurator is also available from the Microsoft Store**)!

For those dinosaurs of us running GPO locally it's a bit too slow to manage the lists via [AGPM](https://learn.microsoft.com/en-us/microsoft-desktop-optimization-pack/agpm/) (check out, change, check in, ask another department for permission to deploy, wait for the other department to validate/deploy, wait for the clients to update...
...and since the lists/mods go hand in hand I think it's better to have both in a file system somewhere.

The lists/mods managment should be swift in testing/deploying with in our case +100 apps (out of 600) possible to convert from traditional static SCCM applications to [winget-installe](https://github.com/Romanitho/Winget-Install)d ones deployed via SCCM to Software Center as available applications upgraded by **WAU**

But, it's possible to implement it here too!

**Reply by KnifMelti** [reply]
Thanks for the idea @Weatherlights! https://github.com/Romanitho/Winget-AutoUpdate/pull/263#issuecomment-1396304079

**Reply by Romanitho** [reply]
should definitely be added to WAU 👏

**Comment by Romanitho** [comment]
Hi, quick question regarding GPO. I noticed that you copy the registry settings of the GPO to WAU registry settings. Wouldn't it be easier to just load default WAU settings, then override them with the GPO registry settings ? instead of copy/remove and stuff ?
This way, the Get-Policies function would become not necessary

**Reply by KnifMelti** [reply]
> Hi, quick question regarding GPO. I noticed that you copy the registry settings of the GPO to WAU registry settings. Wouldn't it be easier to just load default WAU settings, then override them with the GPO registry settings ? instead of copy/remove and stuff ? This way, the Get-Policies function would become not necessary

I don't follow you, load defaults in where?
Override how if not by  comparing/copying?
GPO settings are dynamic and are changed from time to time, thus the compare
And in the function I create/delete links,  change tasks etc; not just changes in the registry...

**Reply by Romanitho** [reply]
In fact, at the moment I'm thinking about optimisation etc. But i understand the thing with the tasks 👌🏼

**Reply by KnifMelti** [reply]
Tested with (no changed setting in GPO):
```
#Get and set Domain/Local Policies (GPO)
$sw = [Diagnostics.Stopwatch]::StartNew()
$ActivateGPOManagement, $ChangedSettings = Get-Policies
$sw.Stop()
Write-ToLog $sw.Elapsed.TotalMilliseconds
```
Result:
20:17:31 - Running in System context
20:17:31 - **56.5689** `= 0,0565689 seconds`

**Reply by Romanitho** [reply]
perfect

**Reply by Romanitho** [reply]
What I wanted to say for GPO is that we have "same" registry keys in `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate` and `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate`. So we could load "default" config and override it with "Policies" ones. This would reduce lines of code in "get-policies" and probably much more.
It's just an idea, as I didn't really parse all your code yet, but it's an interesting part where I start to work with :)

**Reply by Romanitho** [reply]
I started to test something on my test branch

**Reply by KnifMelti** [reply]
Nice if you can find another solution, must admit I've lost track of how I managed to write a function that works as it should.
Too complex...

**Reply by Romanitho** [reply]
yes, it's quite complexe. Mainly when it concerns scheduled task and shortcuts. Maybe we should have kept that out of GPO and change settings only on installation/re-installation. But regarding pure registry keys, it is quite simple to handle finally. We just have to load `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate` then overwrite values with `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate`.
In this way, even next configuration settings that can be implemented in WAU in the futur can be configure with GPO without big extra work :)
My English is quite limited, but I hope you got me haha

**Reply by KnifMelti** [reply]
Yes, I agree. But I think all setting should be available from within GPO to change when needed.
When you're at it, could you explain in the Readme.md under Policies how you import the files to Intune?

**Reply by Romanitho** [reply]
Yes I can :)

**Reply by Romanitho** [reply]
But first, I want to be sure it works 😅

**Comment by Romanitho** [comment]
#418
